src/transformers/trainer_utils.py

```diff
@@ -263,13 +263,14 @@ def total_processes_number(local_rank):
         import torch_xla.core.xla_model as xm
 
         return xm.xrt_world_size()
-    elif is_sagemaker_dp_enabled():
-        import smdistributed.dataparallel.torch.distributed as dist
-
-        return dist.get_world_size()
+    elif is_sagemaker_dp_enabled() and is_torch_available():
+        import torch
+        import smdistributed.dataparallel.torch.torch_smddp
+        torch.distributed.init_process_group(backend="smddp")
```

Comment:
The call to the `init_process_group` should only be done once in the `TrainingArguments`.

Reply:
updated, then we don't need extra logic for SMDP since process group is already initialized and using SMDP backend.